apps/taxes/src/modules/taxes/distribute-discount.ts

```diff
@@ -0,0 +1,24 @@
+import { numbers } from "./numbers";
+
+// ? shouldn't it be used in all providers?
+
+export function distributeDiscount(discountSum: number, prices: number[]) {
+  const totalNumbers = prices.length;
+  const totalSum = prices.reduce((sum, number) => sum + number, 0);
+
+  if (discountSum > totalSum) {
+    throw new Error("Discount cannot be greater than total sum");
+  }
+
+  const discountRatio = discountSum / totalSum;
```

Comment:
And it might be not an artificial situation - some people sell 3d designs and offer free downloads in the same shop. 

For example https://www.alch.shop/shop/p/planter-mold

Reply:
Yes, but would we throw an error then, or just return no discounts?